src/python_testing/TC_OPSTATE_2_3.py

```diff
@@ -80,74 +80,181 @@ async def test_TC_OPSTATE_2_3(self):
+            self.print_step(9, "Read CountdownTime attribute")
             countdown_time = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.CountdownTime)
-            logging.info("CountdownTime: %s" % (countdown_time))
+            logging.info("CountdownTime: %s" % countdown_time)
             asserts.assert_true(countdown_time != 0 or countdown_time == NullValue,
                                 "invalid CountdownTime(%s). Must be a non zero integer, or null" % countdown_time)
             asserts.assert_equal(countdown_time, initial_countdown_time, "CountdownTime(%s) does not equal to the intial CountdownTime(%s)"
                                  % (countdown_time, initial_countdown_time))
 
-        self.print_step(9, "Send Pause command")
+        self.print_step(10, "Send Pause command")
         ret = await self.send_pause_cmd()
         asserts.assert_equal(ret.commandResponseState.errorStateID, Clusters.OperationalState.Enums.ErrorStateEnum.kNoError,
                              "errorStateID(%s) should be NoError(0x00)" % ret.commandResponseState.errorStateID)
 
-        self.print_step(10, "Send Resume command")
+        self.print_step(11, "Send Resume command")
         ret = await self.send_resume_cmd()
         asserts.assert_equal(ret.commandResponseState.errorStateID, Clusters.OperationalState.Enums.ErrorStateEnum.kNoError,
                              "errorStateID(%s) should be NoError(0x00)" % ret.commandResponseState.errorStateID)
 
-        self.print_step(11, "Read OperationalState attribute")
+        self.print_step(12, "Read OperationalState attribute")
         operational_state = await self.read_mod_attribute_expect_success(endpoint=self.endpoint,
                                                                          attribute=attributes.OperationalState)
-        logging.info("OperationalState: %s" % (operational_state))
-        asserts.assert_equal(operational_state, Clusters.OperationalState.Enums.OperationalStateEnum.kRunning,
-                             "OperationalState(%s) should be Running(0x01)" % operational_state)
+        logging.info("OperationalState: %s" % operational_state)
+        asserts.assert_equal(operational_state, old_opstate_dut,
+                             "OperationalState(%s) should be the state before pause (%s)" % operational_state, old_opstate_dut)
 
-        self.print_step(12, "Send Resume command")
+        self.print_step(13, "Send Resume command")
         ret = await self.send_resume_cmd()
         asserts.assert_equal(ret.commandResponseState.errorStateID, Clusters.OperationalState.Enums.ErrorStateEnum.kNoError,
                              "errorStateID(%s) should be NoError(0x00)" % ret.commandResponseState.errorStateID)
 
-        self.print_step(13, "Manually put the device in a state where it cannot receive a Pause command")
+        if self.check_pics("OPSTATE.S.M.RESUME_AFTER_ERR"):
```

Comment:
This PR makes changes to the wrong file :) 

There's a TC 2.3 test script for the **(base) OpState** cluster which needs small changes - see steps 15-18 in the test plan PR: https://github.com/CHIP-Specifications/chip-test-plans/pull/3750/files

There's also a TC 2.3 script for the **RVC OpState** cluster, where most of the changes below need to go.

Reply:
I overwrote my previous updates to the opstate tests. I have now fixed this. Please revisit both.